controllers/akodeploymentconfig/cluster/cluster_controller_addon_secret.go

```diff
@@ -57,6 +59,13 @@ func (r *ClusterReconciler) ReconcileAddonSecret(
 		}
 	}
 
+	//Stop reconciling if AKO ip family doesn't match cluster node ip family
+	if err = validateADCAndClusterIpFamily(cluster, obj, isVIPProvider, log); err != nil {
+		log.Error(err, "Error: ip family in AKODeploymentConfig doesn't match cluster node ip family")
+		conditions.MarkTrue(cluster, akoov1alpha1.ClusterIpFamilyValidationFailedCondition)
```

Comment:
Will this update the cluster object?
Better to double-check that

Reply:
I think it will update. We can see this condition in cluster config
```
conditions:
  - lastTransitionTime: "2023-09-07T02:49:42Z"
    reason: WaitingForKubeadmInit
    severity: Info
    status: "False"
    type: Ready
  - lastTransitionTime: "2023-09-07T02:46:33Z"
    status: "True"
    type: ClusterIpFamilyValidationFailed
```